projects/smoke_bts7040/src/main.c

```diff
@@ -0,0 +1,111 @@
+// A simple smoke test for BTS7040
+
+// Periodically take reading from user selected channels and log the result
+// Configurable items: wait time, FRONT or REAR power distro selection, channels to be tested
+
+#include "bts7040_load_switch.h"
+#include "controller_board_pins.h"
+#include "current_measurement_config.h"
+#include "gpio.h"
+#include "i2c.h"
+#include "interrupt.h"
+#include "log.h"
+#include "mux.h"
+#include "pca9539r_gpio_expander.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+// experimentally determined value: scaling factor of 670/1160 ~ 0.578 ohms, bias of -8 mA
+#define SMOKE_BTS7040_SENSE_RESISTOR 1160
+#define SMOKE_BTS7040_BIAS (-8)
+
+#define SMOKE_BTS7040_MIN_FAULT_VOLTAGE_MV 3200
+#define SMOKE_BTS7040_MAX_FAULT_VOLTAGE_MV 10000
+
+// Smoke test settings. Can be modified to fit testing purpose.
+#define CURRENT_MEASURE_INTERVAL_MS 500  // Set wait time between each set of readings
+#define IS_FRONT_POWER_DISTRO true       // Set whether to test FRONT or REAR power distro
+
+// Maximum number of channels that can be tested,
+// which is the same number of bts7200 used in front/rear power distribution
+#define MAX_TEST_CHANNELS 8
+
+#define I2C_PORT I2C_PORT_2
+#define PIN_I2C_SCL CONTROLLER_BOARD_ADDR_I2C2_SCL
+#define PIN_I2C_SDA CONTROLLER_BOARD_ADDR_I2C2_SDA
+
+// Set of channels to be tested, the array contains the indices of the BTS7200 in the front/rear HW
+// config Details at
+// https://uwmidsun.atlassian.net/wiki/spaces/ELEC/pages/1419740028/BTS7200+smoke+test+user+guide
+static uint8_t s_test_channels[] = { 0, 1, 2, 3, 4, 5, 6, 7 };
```

Comment:
* the array should not contain 6 or 7 because there are only 6 channels (indices in the bts7040s arrays)
* change `BTS7200` to `BTS7040` in the comment
* the linked Confluence page is for the BTS7200 smoke test and says which channel (index) corresponds to each bts7200. It might be a good idea to make an analogous page for this smoke test and link it instead, but the incorrect link should definitely be removed to avoid confusion.